docs/app-integration-development/08-OAuth-2-Functionality.md

```diff
@@ -51,5 +51,119 @@ There two options for implementing PagerDuty OAuth in your app. [PKCE (Proof Key
 
 **Server-side App* - an app running on a server which can securely store secrets
 
+## Cracking Open an ID Token
+
+Pagerduty uses ID tokens to fetch access tokens, which are structured as [Json Web Tokens (JWT)](https://datatracker.ietf.org/doc/html/rfc7519). The token is encoded and signed, and contains claims from the OpenID Connect Provider used. 
+
+The main parts of the ID token are the following.
+
+  Name            | Description
+----------------- | -----------
+`Header`          | Contains token metadata such as the type of token and hash used
+`Payload`         | Contains user, authorization, and claims
+`Signature`       | Used to verify the token and check for tampering
+
+Each part of the token is encoded using the algorithm specified in the header, split by `.`. Every part is structured in the JSON format.
+
+### Header
+The main purpose of the header is to determine a decoding algorithm to use for the rest of the token and what key is used for validation. All valid header parameters can be found [here](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1), but the most common ones are the following.
+
+  Claim           | Description
+----------------- | -----------
+`kid`             | Key ID found on the JWKS endpoint of the issuer
+`x5t`             | Fingerprint of the certificate used
+`alg`             | Algoirthm used for signature of the token
+
+### Payload
+The payload contains all claims, which are statements about the user along with any additional data needed. There are three types of claims, and are the following.
```

Comment:
this wording is a little awkward. maybe 'The three types of claims are as follows'?